src/QMCHamiltonians/EnergyDensityEstimator.h

```diff
@@ -35,7 +35,7 @@ class EnergyDensityEstimator : public OperatorBase, public PtclOnLatticeTraits
   Return_t evaluate(ParticleSet& P);
   void addObservables(PropertySetType& plist) {}
   void addObservables(PropertySetType& plist, BufferType& olist);
-  void registerCollectables(std::vector<observable_helper*>& h5desc, hid_t gid) const;
+  void registerCollectables(std::vector<observable_helper>& h5desc, hid_t gid) const;
```

Comment:
Since the API changed, it can be a chance to add override keyword on all the derived classes. It is your call to change directly in this PR or do it separately. I don't have a preference just want to point this out.

Reply:
@ye-luo good point. My preference is to try to not use `override` as it breaks the `is-a` relationship. This case is simple and the derived classes should be compliant with the base signature. Thanks!

Reply:
I don't get your point `it breaks the is-a relationship`. My understand of `override` is that it brings clarity and prevent developer errors.

Reply:
@ye-luo you are right on `override`. I totally mispoke I meant not using it. I'd still prefer using `final` instead where applicable, thoughts?

Reply:
Although my experience with `final` is limited, I feel it only useful to scan inheritance by the compiler.
```
class base
{
  virtual void foo() = 0;
}

class A: public Base
{
  virtual void foo() override { //1st level implementation } ;
}

class B: public A
{
  void foo() override { //2nd level implementation } ;
}
```
in this case, final cannot be added to `A::foo`.

final is stronger than override. The additional restriction from final cannot be easily justified by only knowing the base class abstraction.

Reply:
Your example is correct `final` would only apply to B (or C if it exist) as a final leaf in the vtable.`B::foo() final` would kill the ambiguity of `B::foo() override` as we would mark it as "final function that can't we can't override", always executed if B is created. As for A (or B if C exists), `override` is the way to go as it is clear: "overrides `Base:foo()`, but it is not final. If C is to be final, we move `final``C::foo()` and redefine to `B:foo() override`. On a related note, some compilers (PGI if I recall correctly) issue warnings by default if not all virtual functions are overridden.

Reply:
I would like to see at least override used everywhere, final where appropriate. I think that we have had one error that would have been caught through a required override statement (error not warning). No need to add here though. We have been gradually adding override and could add a mention to the developer docs. I think it certainly brings clarity.

Reply:
In practice I've found override to be quite useful at catching API issues during development. It hasn't made like the amount of virtual inheritance in the code more but its alerted me to quite a number of potential issues very early in coding rather than at runtime.